### PR TITLE
Add explicit {{ stdlib(c) }} build dependency

### PR DESCRIPTION
**Pull request by peterjc**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by peterjc** [comment]
@conda-forge-admin, please rerender

**Comment by peterjc** [comment]
Windows build is failing;

```
cl.exe -O2 -Wall -std=c99 -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64   -c -o flash.o flash.c
Microsoft (R) C/C++ Optimizing Compiler Version 19.40.33811 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

cl : Command line warning D9035 : option 'o' has been deprecated and will be removed in a future release
cl : Command line warning D9002 : ignoring unknown option '-std=c99'
flash.c
flash.c(31): fatal error C1083: Cannot open include file: 'getopt.h': No such file or directory
mingw32-make: *** [flash.o] Error 2
<builtin>: recipe for target 'flash.o' failed
```

**Comment by peterjc** [comment]
Adding ``m2w64-toolchain` didn't work. Will revert that along with any alternative idea...

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,13 +10,15 @@ source:
...
-  number: 0
+  number: 1
 
 requirements:
   build:
     - make
     - {{ compiler('c') }}  # [unix]
     - {{ compiler('m2w64_c') }}  # [win]
+    - m2w64-toolchain  # [win]
+    - {{ stdlib("c") }}
```
Comment (h-vetinari) [inline recipe/meta.yaml]:
```suggestion
    - {{ compiler('c') }}        # [unix]
    - {{ stdlib('c') }}          # [unix]
    - {{ compiler('m2w64_c') }}  # [win]
    - {{ stdlib('m2w64_c') }}    # [win]
```

**Review by h-vetinari — COMMENTED** [review]
`getopt.h` is not a header [covered](https://en.cppreference.com/w/c/header) by the C standard, but rather [part of](https://en.wikipedia.org/wiki/Getopt) POSIX. You won't find it in the windows default stdlib, which you're using here. You can try the following.

File: recipe/meta.yaml (hunk 1 above)
Comment (h-vetinari) [reply recipe/meta.yaml]: You then also have to rerender the feedstock, see docs for [bot commands](https://conda-forge.org/docs/maintainer/infrastructure/#admin-web-services).

@conda-forge-admin, please rerender

**Comment by peterjc** [comment]
Thanks. I'm assuming this worked 3 years ago with an implicit dependency when built on Windows.

**Comment by peterjc** [comment]
@h-vetinari nice! Does this now look good to merge to you?

**Pull request merged**